master/internal/api_command.go

```diff
@@ -241,7 +246,25 @@ func (a *apiServer) GetCommand(
 func (a *apiServer) KillCommand(
 	ctx context.Context, req *apiv1.KillCommandRequest,
 ) (resp *apiv1.KillCommandResponse, err error) {
-	if _, err := a.GetCommand(ctx, &apiv1.GetCommandRequest{CommandId: req.CommandId}); err != nil {
+	targetCmd, err := a.GetCommand(ctx, &apiv1.GetCommandRequest{CommandId: req.CommandId})
+	if err != nil {
+		return nil, err
+	}
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = command.AuthZProvider.Get().CanTerminateNSC(
+		ctx, *curUser, model.AccessScopeID(targetCmd.Command.WorkspaceId),
+	)
+	err = apiutils.MapAndFilterErrors(err, nil, nil)
```

Comment:
this is kind of a common pattern and we might want to add more logic here about sometimes reporting perm errors as 404 and sometimes expose the  permission issue.

Reply:
Wouldn't it be 404 errors for listing commands but permission issue for other things? Maybe it's worth having a conversation about, but I'm not sure it returning 404 matters much if they don't have permission to view it anyways?

Reply:
we chatted about this offiline. ref https://determined-ai.slack.com/archives/C03LE5Z8R46/p1672776613155459?thread_ts=1672772755.830529&cid=C03LE5Z8R46